### DISCUSSION TITLE
How the best stream format?

### DISCUSSION THREAD

**Discussion by renatoianhez** [schreibfaul1/ESP32-audioI2S#347] [General]
There are some radio stations on the web that are really concerned with spreading access to their stations. Take for example [Antenne](https://www.antenne.de/programm/empfang/antenne-bayern-hoeren-internet), a set of webradios from Germany. As can be seen in this link, Antenne provides URLs in different formats (m3u, mp3, aac...) for all its radio stations. Which of the formats works best with this library?

**Comment by schreibfaul1** [comment]
That is hard to say. If you are planning a radio application and value short switching times, you should avoid playlists (m3u, pls, asx) and use the direct link (displays the event audio_lasthost()) If possible, avoid encrypted connections (better http instead of https) is faster and uses less memory. mp3 and aac are ok. If you use an ESP32-S3, the SBR (Spectral Band Replication) is automatically switched on for AAC, which improves the sound. Unfortunately, HE-AAC does not work (also aacp, aac+) The AAC decoder does not know 'Parametric Stereo' and switches to mono in compatibility mode. But there are only a few stations that use AAC+
